Enable Substring Search for Flow IDs in MySQL Flow Repository

yashmurthy98 opened the pull request:
Closes https://github.com/kestra-io/kestra/issues/1502.

Added support to search flows with in-between characters in ID.
This support also enables the search with no minimum number of characters.

This PR closes the issue: #1502 

 Steps to test this support:
1. Start your Kestra backend and MySQL database.
2. Ensure you have flows with IDs that contain a common substring.
3. Open the Kestra UI in your browser and navigate to the Flows page in the UI.
4. Type a substring in the search bar.
5. Flows with IDs containing the substring should appear in the results

yashmurthy98 committed:
Search by contains on ID for MySQL repo

tchiotludo reviewed (changes requested):
thanks @yashmurthy98 for the contribution, but we can't accept this pull request since like query is not using any index and will be break the whole performance of the application. 

Why we built a full text index to have a proper index, and a common way to manage research on the application. 

Maybe a fine tuning of the full text index could be possible ?

yashmurthy98 commented:
Thanks @tchiotludo for the feedback. 
I will check how full text index could be fine tuned to support sub string searches without LIKE query.

yashmurthy98 commented:
Hi @tchiotludo 
Can we use FULLTEXT index with N-gram parser to provide substring search support? 
N-gram parser is supported by MySQL versions starting from version 5.7.6 and storage engine should be InnoDB. Will it impact existing users?

tchiotludo commented:
we are using innodb everywhere so it's not a problems. 
For min version, we advice for mysql 8 as I remember, si it's not an issue.

yashmurthy98 committed:
Searching on substring with N-gram support

yashmurthy98 commented:
Hi @tchiotludo 
I’ve updated the code by removing the LIKE operator-based query and added a Flyway migration SQL file to alter the ix_fulltext FULLTEXT index in MySQL using the ngram parser.

Additionally, I modified the match condition to allow searching for single-character strings.
To support this, we have two options:

1. Use the current approach in code that works with the default ngram_token_size = 2.
2. Alternatively, set ngram_token_size = 1 at the MySQL system level, which may require a change in deployment configuration.
Please let me know which approach is preferred or more feasible for deployment strategy.

Please refer to the demonstration of this functionality here: https://jam.dev/c/f8fa3e7c-7e5c-444d-805e-d09ae7e305fd

yashmurthy98 commented:
Hi @nkwiatkowski 
just checking in, is there anything you’d like me to address? Let me know if any changes are needed. Thanks!

nkwiatkowski reviewed (changes requested):
Hi, sorry for the late review.
I don't know exactly what causes that, but your change doesn't work as the query use to. MySql repositories have 3 failing tests related the query.
Can you fix that please?

yashmurthy98 commented:
I will check the unit tests and investigate why the difference in functionality. Thanks for the update!

yashmurthy98 committed:
Merge branch 'kestra-io:develop' into flow-search-mysql

yashmurthy98 committed:
Merge branch 'kestra-io:develop' into flow-search-mysql

yashmurthy98 committed:
Merge branch 'kestra-io:develop' into flow-search-mysql

MilosPaunovic commented:
Hey @yashmurthy98,

Have you managed to handle failing tests yet?

MilosPaunovic committed:
Merge branch 'develop' into flow-search-mysql

yashmurthy98 commented:
Hi @MilosPaunovic
I apologise for the delay in response. I am working on the failing tests. I will update the PR as soon as the tests are resolved.

Thanks

yashmurthy98 committed:
Merge branch 'kestra-io:develop' into flow-search-mysql

yashmurthy98 committed:
Merge branch 'kestra-io:develop' into flow-search-mysql

MilosPaunovic commented:
Hey @yashmurthy98, have you managed to have a look at the failing tests?

loicmathieu commented:
Closing this PR as we haven't heard back from the contributor for 2 months.
Please don't hesitate to reopen it when all feedback has been taken care of, if you still want to work on it.
